src/libs/AppStateMonitor/index.js

```diff
@@ -0,0 +1,35 @@
+import {AppState} from 'react-native';
+import CONST from '../../CONST';
+import shouldReportActivity from './shouldReportActivity';
+
+let appState = CONST.APP_STATE.ACTIVE;
+
+/**
+ * Listener that will only fire the callback when the user has become active.
+ *
+ * @param {Function} callback
+ * @returns {Function} to unsubscribe
+ */
+function addBecameActiveListener(callback) {
+    /**
+     * @param {String} state
+     */
+    function appStateChangeCallback(state) {
+        if (
+            shouldReportActivity
+            && (appState === CONST.APP_STATE.INACTIVE || appState === CONST.APP_STATE.BACKGROUND)
+            && state === CONST.APP_STATE.ACTIVE
```

Comment:
I think I understand this but basically this is only change the app state if the user has become active and the stored state is not active right?

Reply:
Correct. We set up the state (assume to be active when the app inits) then listen for changes and save it locally each time so we can detect a change from one state to another.